### DISCUSSION TITLE
Create MultipleChooserPanel records programmatically in Wagtail

### DISCUSSION THREAD

**Discussion by htabar** [wagtail/wagtail#12219] [Q&A]
Hi everyone,

I've been following the guidance from [this StackOverflow answer](https://stackoverflow.com/a/77433742), and everything works perfectly in the admin interface—I can manually add, edit, and remove ArticleAuthor records using a MultipleChooserPanel.

However, I'm facing an issue when trying to add ArticleAuthor records programmatically. Although I can create the objects, they don't appear in the admin interface. I suspect this might be related to how Django's formsets operate.

Could anyone advise on the correct approach to ensure that records created programmatically show up in the MultipleChooserPanel as if they were added manually through the admin interface?

Any guidance would be greatly appreciated.

Thanks!

**Comment by ababic** [comment]
@htabar out of interest, were you doing this for a page, or some other model with drafts? If you write directly to live data tables, those changes won't be automically reflected in pre-existing drafts (which is what gets loaded into the editor when you edit the page in Wagtail). You have a couple of choices...

1. Engage with the revisions system
2. Delete all revisions for the page in question, so that Wagtail loads the 'live' representation into the editor

**Comment by htabar** [comment]
Thanks @ababic for reaching out! We ended up using InlinePanel instead of MultipleChooserPanel, since we only needed to select up to 2 or 3 items.

Also, our model was a Snippet, not a Page, but it inherited from DraftStateMixin and RevisionMixin so that the views behaved similarly to Page models. We later removed those mixins because they added unnecessary complexity for our use case.

I didn’t try your recommendations, but I believe our issue was indeed related to drafts, as you mentioned — and I suspect your suggestion would have resolved it (though I haven’t tested it yet).
